### PR TITLE
[Logging] Search

### PR DESCRIPTION
**Pull request by osamasayed**
This PR adds logging to:

1. Command bar
2. Search drawer
3. Search page

### CODE REVIEW

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**/search**](https://quran-com-abqeyvybq-qurancom.vercel.app/search) |
|:-:|
|<a href="https://quran-com-abqeyvybq-qurancom.vercel.app/search">Screenshot of /search</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Yv9LIZCLo9NLdMWRZrDCuD-g-IEruEFevLS-nZzIsJvKs5Wk46cxhluanSUuPq5HBKvoKRO4JzzOFC8-1oLvsg.png">(view full size)</a> |







Commit cc36610505217edc46b4a96c2aff3fb515fe4e1a (https://quran-com-abqeyvybq-qurancom.vercel.app).

File: src/components/Navbar/SearchDrawer/Footer/index.tsx [hunk 1]
```diff
@@ -24,7 +25,14 @@ const Footer: React.FC = () => {
           </div>
           <div className={classNames(styles.beta, styles.uppercase)}>{t('beta')}</div>
         </div>
-        <Link href="https://download.tarteel.ai" newTab variant={LinkVariant.Highlight}>
+        <Link
+          href="https://download.tarteel.ai"
+          newTab
+          variant={LinkVariant.Highlight}
+          onClick={() => {
+            logButtonClick('search_drawer_footer_learn_tarteel');
```
Comment (tifa2UP) [inline src/components/Navbar/SearchDrawer/Footer/index.tsx:33]: nit `search_drawer_footer_tarteel_attribution`

File: src/hooks/useTarteelVoiceSearch.ts [hunk 2]
```diff
@@ -71,6 +72,7 @@ const useTarteelVoiceSearch = (startRecording = true) => {
     ) => {
       // if the websocket is still open, close it
       if (isWebSocketOpen(webSocket)) {
+        logEvent('tarteel_websocket_close');
```
Comment (tifa2UP) [inline src/hooks/useTarteelVoiceSearch.ts:75]: this is probably not important to log?

File: src/hooks/useTarteelVoiceSearch.ts [hunk 3]
```diff
@@ -188,12 +195,14 @@ const useTarteelVoiceSearch = (startRecording = true) => {
             new AudioWorklet(new URL('src/audioInput/MicInputProcessor.ts', import.meta.url)),
           )
           .then(() => {
+            logEvent('tarteel_websocket_initialize');
```
Comment (tifa2UP) [inline src/hooks/useTarteelVoiceSearch.ts:198]: ditto, don't think is important

File: src/hooks/useTarteelVoiceSearch.ts [hunk 4]
```diff
@@ -188,12 +195,14 @@ const useTarteelVoiceSearch = (startRecording = true) => {
...
             setIsLoading(true);
             // 3. Start a new websocket
             webSocket = new WebSocket(
               `wss://voice-v2.tarteel.io/search/?Authorization=${process.env.NEXT_PUBLIC_TARTEEL_VS_API_KEY}`,
             );
             webSocket.onopen = () => {
+              logEvent('tarteel_websocket_open');
```
Comment (tifa2UP) [inline src/hooks/useTarteelVoiceSearch.ts:205]: ditto, don't think is important

**Review by tifa2UP — APPROVED** [review]

**Pull request merged**